lib/child_process.js

```diff
@@ -145,16 +145,12 @@ exports.execFile = function(file /*, args, options, callback*/) {
     windowsVerbatimArguments: !!options.windowsVerbatimArguments
   });
 
-  var encoding;
-  var _stdout;
-  var _stderr;
+  var encoding, stdoutEncoding, stderrEncoding;
```

Comment:
Hmmm...It's possible that the encoding could change after data has already emitted, right? Like, one chunk is utf-8 but the next is utf-16 or whatever? If that's the case, it's probably not enough to store the encoding for the entire stream; rather, it has to be stored for each chunk?